src/app/shared/components/message-monitor/message-monitor.component.ts

```diff
@@ -87,6 +87,21 @@ export class MessageMonitorComponent implements OnInit, OnChanges, OnDestroy {
       return;
     }
 
+    switch (this.valType) {
+      case 'string':
+        this.filters.stringValue = this.valHttp;
+        break;
+      case 'data':
+        this.filters.dataValue = this.valHttp;
+        break;
+      case 'bool':
+        this.filters.boolValue = this.valHttp;
+        break;
+      case 'float':
+        this.filters.value = this.valHttp;
+        break;
+    }
+
```

Comment:
You can take advantage of the javascript keys and do this in one line. `this.filters[this.valType], where `this.valType` is one of the switch cases.